### ISSUE TITLE
Periodic panic error during playback on cache

### ISSUE THREAD

**Issue by Krandor1** [ncw/rclone#2448]
This is the second time I've seen this and grabbed the debug.  I was playing a file via plex and it stopped partway though.  Tried to retry that one and another one and they both failed.  Here is the log from a playback.    Last time I had this happpen only fix was to tell rclone to rebuild the cache.  

```
2018/08/06 19:57:03 DEBUG : &{tv/Newhart/Season 01/Newhart - 01x08 - Some are Born Writers.Others Have Writers Thrust Upon Them DVD SPRiNTER.mkv (r)}: >Read: read=0, err=<nil>
2018/08/06 19:57:03 fuse: panic in handler for Read [ID=0xb7a805 Node=0x3d0a Uid=109 Gid=115 Pid=21143] 0xc 131072 @0x0 dir=false fl=0 lock=0 ffl=OpenReadOnly: runtime error: invalid memory address or nil pointer dereference
goroutine 12381933 [running]:
github.com/ncw/rclone/vendor/bazil.org/fuse/fs.(*Server).serve.func2(0x11b5a80, 0xc4476c7490, 0xc44366def8, 0xc44366de6f)
        /home/travis/gopath/src/github.com/ncw/rclone/vendor/bazil.org/fuse/fs/serve.go:857 +0x1b1
panic(0xf19140, 0x17d3cd0)
        /home/travis/.gimme/versions/go1.10.1.linux.amd64/src/runtime/panic.go:502 +0x229
github.com/ncw/rclone/backend/cache.(*plexConnector).authenticate(0xc4200e43f0, 0x0, 0x0)
        /home/travis/gopath/src/github.com/ncw/rclone/backend/cache/plex.go:213 +0x639
github.com/ncw/rclone/backend/cache.(*Handle).startReadWorkers(0xc44001bea0)
        /home/travis/gopath/src/github.com/ncw/rclone/backend/cache/handle.go:105 +0xdd
github.com/ncw/rclone/backend/cache.NewObjectHandle(0xc44b8aef00, 0xc4204b61c0, 0x0)
        /home/travis/gopath/src/github.com/ncw/rclone/backend/cache/handle.go:77 +0x151
github.com/ncw/rclone/backend/cache.(*Object).Open(0xc44b8aef00, 0xc436f04ab0, 0x1, 0x1, 0x1, 0xc436f04ab0, 0x0, 0x1)
        /home/travis/gopath/src/github.com/ncw/rclone/backend/cache/object.go:216 +0x67
github.com/ncw/rclone/backend/crypt.(*Object).Open.func1(0x0, 0x8008020, 0x18, 0x28, 0xc4618eed50, 0x0)
        /home/travis/gopath/src/github.com/ncw/rclone/backend/crypt/crypt.go:688 +0xf1
github.com/ncw/rclone/backend/crypt.(*cipher).newDecrypterSeek(0xc420354be0, 0xc4618eed50, 0x0, 0x8000000, 0xf88aa0, 0x1, 0xc4618eed50)
        /home/travis/gopath/src/github.com/ncw/rclone/backend/crypt/cipher.go:767 +0xd5
github.com/ncw/rclone/backend/crypt.(*cipher).DecryptDataSeek(0xc420354be0, 0xc4618eed50, 0x0, 0x8000000, 0xc421416800, 0xc42b62d220, 0x7f9eb55e91e8, 0x0)
        /home/travis/gopath/src/github.com/ncw/rclone/backend/crypt/cipher.go:1045 +0x4d
github.com/ncw/rclone/backend/crypt.(*Object).Open(0xc42b62d220, 0xc436f04aa0, 0x1, 0x1, 0x0, 0x0, 0x2, 0x4100a9)
        /home/travis/gopath/src/github.com/ncw/rclone/backend/crypt/crypt.go:674 +0x350
github.com/ncw/rclone/fs/chunkedreader.(*ChunkedReader).openRange(0xc4207745a0, 0x10f79f8, 0xc4207745a0)
        /home/travis/gopath/src/github.com/ncw/rclone/fs/chunkedreader/chunkedreader.go:220 +0x3aa
github.com/ncw/rclone/fs/chunkedreader.(*ChunkedReader).Open(0xc4207745a0, 0x0, 0x0, 0x0)
        /home/travis/gopath/src/github.com/ncw/rclone/fs/chunkedreader/chunkedreader.go:181 +0x89
github.com/ncw/rclone/vfs.(*ReadFileHandle).openPending(0xc491414cb0, 0xc4232938f0, 0x1)
        /home/travis/gopath/src/github.com/ncw/rclone/vfs/read.go:68 +0xf4
github.com/ncw/rclone/vfs.(*ReadFileHandle).readAt(0xc491414cb0, 0xc47a704000, 0x20000, 0x20000, 0x0, 0xe9a9c0, 0x1, 0xc47a704000)
        /home/travis/gopath/src/github.com/ncw/rclone/vfs/read.go:199 +0x43
github.com/ncw/rclone/vfs.(*ReadFileHandle).ReadAt(0xc491414cb0, 0xc47a704000, 0x20000, 0x20000, 0x0, 0x0, 0x0, 0x0)
        /home/travis/gopath/src/github.com/ncw/rclone/vfs/read.go:194 +0xa1
github.com/ncw/rclone/cmd/mount.(*FileHandle).Read(0xc436f049c0, 0x11b7440, 0xc42a2a1b00, 0xc4476c7490, 0xc480b9a540, 0x20001, 0x5d)
        /home/travis/gopath/src/github.com/ncw/rclone/cmd/mount/handle.go:28 +0x26b
github.com/ncw/rclone/vendor/bazil.org/fuse/fs.(*Server).handleRequest(0xc420160240, 0x11b7440, 0xc42a2a1b00, 0x11ab0a0, 0xc43526d808, 0xc42a2a1a40, 0x11b5a80, 0xc4476c7490, 0xc44366def8, 0x20796c696d61462f, ...)
        /home/travis/gopath/src/github.com/ncw/rclone/vendor/bazil.org/fuse/fs/serve.go:1249 +0x8c8
github.com/ncw/rclone/vendor/bazil.org/fuse/fs.(*Server).serve(0xc420160240, 0x11b5a80, 0xc4476c7490)
        /home/travis/gopath/src/github.com/ncw/rclone/vendor/bazil.org/fuse/fs/serve.go:878 +0x41e
github.com/ncw/rclone/vendor/bazil.org/fuse/fs.(*Server).Serve.func1(0xc420160240, 0x11b5a80, 0xc4476c7490)
        /home/travis/gopath/src/github.com/ncw/rclone/vendor/bazil.org/fuse/fs/serve.go:425 +0x6e
created by github.com/ncw/rclone/vendor/bazil.org/fuse/fs.(*Server).Serve
        /home/travis/gopath/src/github.com/ncw/rclone/vendor/bazil.org/fuse/fs/serve.go:423 +0x331
2018/08/06 19:57:03 DEBUG : &{tv/Newhart/Season 01/Newhart - 01x08 - Some are Born Writers.Others Have Writers Thrust Upon Them DVD SPRiNTER.mkv (r)}: Read: len=4096, offset=0
2018/08/06 19:57:03 DEBUG : tv/Newhart/Season 01/Newhart - 01x08 - Some are Born Writers.Others Have Writers Thrust Upon Them DVD SPRiNTER.mkv: ChunkedReader.openRange at 0 length 134217728
```

```
root@voldemort:/home/bh# rclone -V
rclone v1.42-067-gffd11662β
- os/arch: linux/amd64
- go version: go1.10.1
```

**Comment by ncw** [comment]
This looks like the `saveToken` function is `nil` here

https://github.com/ncw/rclone/blob/d95667b06d85f68a01de4ae4ac643343e5320ef9/backend/cache/plex.go#L213

Not sure exactly why that might be though... I guess it is possible if newPlexConnectorWithToken was called.

What do you think @remusb ?

**Comment by ncw** [comment]
PS and thanks for reporting this @Krandor1

**Comment by Krandor1** [comment]
So based on your comment I went and checked another log and it does look like plex restarted right around the time I started to see this.  I have an app I run (monit) which monitors plex and restarts it if needed and that did kick off right before this happened.  Here are those logs from 10 minutes before the logs above and when this started I probably fiddeled with it 5-10 minutes on plex before pulling up the rlcone logs above so calling newPlexConnectorWithToken sounds reasonable due to a restart of plex and so it looks like rclone didn't handle that well.  Also based on this, plex didn't appear to stop cleanly either which could have something to do with it.  

[CDT Aug  6 19:43:42] error    : 'plex' failed protocol test [HTTP] at [localhost]:32400/web/index.html [TCP/IP SSL] -- HTTP: Error receiving data -- Resource temporarily unavailable
[CDT Aug  6 19:43:42] info     : 'plex' trying to restart
[CDT Aug  6 19:43:42] info     : 'plex' stop: /bin/systemctl
[CDT Aug  6 19:44:12] error    : 'plex' failed to stop (exit status -1) -- Program /bin/systemctl timed out
[CDT Aug  6 19:46:12] error    : 'plex' process is not running
[CDT Aug  6 19:46:12] info     : 'plex' trying to restart
[CDT Aug  6 19:46:12] info     : 'plex' start: /bin/systemctl
[CDT Aug  6 19:46:12] info     : 'plex' started
[CDT Aug  6 19:48:12] info     : 'plex' process is running with pid 20854
[CDT Aug  6 19:48:12] info     : 'plex' connection succeeded to [localhost]:32400/web/index.html [TCP/IP SSL]
